cmd/erigon-cl/cl-core/syncfromcheckpoint.go

```diff
@@ -110,3 +110,63 @@ func getStatusSRFn(ctx context.Context, opts sentinelRequestOpts) {
 	}
 	opts.respChan <- resp
 }
+
+func GetBlocksByRange(ctx context.Context, sc consensusrpc.SentinelClient, start, numBlocks uint64) ([]*cltypes.SignedBeaconBlockBellatrix, error) {
+	var i uint64
+
+	totalBlocks := []*cltypes.SignedBeaconBlockBellatrix{}
+	// Fetch blocks in batches of 10 to avoid hitting max message size.
+	for i = 0; i < numBlocks; i += 10 {
+		req := cltypes.BeaconBlocksByRangeRequest{
+			StartSlot: start + i,
+			Count:     10,
+		}
+
+		// Send request
+		resp, err := sendSentinelRequest(ctx, getBlocksByRangeSRFn, sentinelRequestOpts{
+			req: &req,
+			sc:  sc,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("unexpected error sending sentinel request: %v", err)
+		}
+		ret, ok := resp.([]*cltypes.SignedBeaconBlockBellatrix)
+		if !ok {
+			return nil, fmt.Errorf("unable to cast response to type: %+v", resp)
+		}
+		for j := 0; j < len(ret); j++ {
+			totalBlocks = append(totalBlocks, ret[j])
+		}
+	}
+	return totalBlocks, nil
+}
+
+func getBlocksByRangeSRFn(ctx context.Context, opts sentinelRequestOpts) {
+	// Assert that the request the correct type.
+	req := opts.req.(*cltypes.BeaconBlocksByRangeRequest)
+
+	log.Info("Getting blocks by range", "num blocks", req.Count)
+	log.Info("Getting blocks by range", "start slot", req.StartSlot)
+
+	// Request for blocks by root given the finalized checkpoint root.
+	resp, err := rpc.SendBeaconBlocksByRangeReq(ctx, req.StartSlot, req.Count, opts.sc)
+	if err != nil {
+		opts.errChan <- err
+		return
+	}
+	if len(resp) != int(req.Count) {
+		opts.errChan <- fmt.Errorf("unexpected response length, got %d want %d", len(resp), int(req.Count))
+		return
+	}
+	result := make([]*cltypes.SignedBeaconBlockBellatrix, len(resp))
+	// Type assert that the returned object are the correct type.
+	for _, obj := range resp {
+		block, ok := obj.(*cltypes.SignedBeaconBlockBellatrix)
+		if !ok {
+			log.Error("unable to cast object into block", "object", obj)
+			return
+		}
+		result = append(result, block)
```

Comment:
you append to non-empty slice. likely you did mean: `make([]*cltypes.SignedBeaconBlockBellatrix, 0, len(resp))`